raiden/transfer/node.py

```diff
@@ -418,6 +420,22 @@ def handle_token_network_action(node_state, state_change):
     return TransitionResult(node_state, events)
 
 
+def handle_delivered(node_state, state_change):
+    # TODO: imporve the complexity of this algorithm
```

Comment:
What exactly do you mean by this comment? Improve the complexity in what respect?